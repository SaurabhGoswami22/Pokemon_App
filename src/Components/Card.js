import React from "react";
const Card = ({ pokemon, loading,infoPokemon}) => {
    return (
        <>
        {
            loading ? <h1>Loading...</h1> :
                pokemon && pokemon.map((item) => {
                    return (
                        <>
                            <div className="card" key={item.id} onClick={()=>infoPokemon(item)}>
                                <h2>{item.id}</h2>
                                <img src={item.sprites.front_default} alt="" />
                                <h2>{item.name}</h2>&nbsp;
                                <h3>Height:{item.height}</h3>&nbsp;
                                <h3>Weight:{item.weight}</h3>
                            </div>
                            
                        </>
                    )
                })
        }

        </>
    )
}
export default Card;