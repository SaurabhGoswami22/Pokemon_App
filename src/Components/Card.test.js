import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Enzyme from 'enzyme';
Enzyme.configure({ adapter: new Adapter() });
describe('Pokemon_app_card', () => {
    it('should render the card component', () => {
      const wrapper = shallow(
        <Card/>
      );
      expect(wrapper).toMatchSnapshot();
    });
});