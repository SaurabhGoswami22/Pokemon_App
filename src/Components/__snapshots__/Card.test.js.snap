// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokemon_app_card should render the card component 1`] = `ShallowWrapper {}`;
